smart-connector-rest-dist/src/test/java/eu/knowledge/engine/rest/api/TestAskAnswerWithGapsEnabled.java

```diff
@@ -0,0 +1,131 @@
+							.of("Knowledge-Base-Id", answerKBId, "Content-Type", "application/json", "Accept", "*/*"));
+					test.expectStatus(200);
+
+					// build the body to answer the request: add handle request ID and dummy data
+					// bindingset
+					JsonObjectBuilder builder = Json.createObjectBuilder();
+					JsonReader jp = Json.createReader(new StringReader(test.getBody()));
+					JsonObject jo = jp.readObject();
+					int handleRequestId = jo.getInt("handleRequestId");
+					builder.add("handleRequestId", handleRequestId);
+					JsonReader jr = Json.createReader(new StringReader(
+							"[{\"a\": \"<https://www.tno.nl/example/Barry>\",\"b\": \"<https://www.tno.nl/example/Jack>\"}]"));
+					JsonArray bs = jr.readArray();
+					builder.add("bindingSet", bs);
+					JsonObject jo2 = builder.build();
+					String body = jo2.toString();
+					System.out.println("Handle an answer to a request with body: " + body);
+
+					// fire the POST handle to execute the answer
+					var test2 = new HttpTester(new URL(url.toString() + "/sc/handle"), "POST", body,
+							Map.of("Knowledge-Base-Id", answerKBId, "Knowledge-Interaction-Id", answerKIId,
+									"Content-Type", "application/json", "Accept", "*/*"));
+					test2.expectStatus(200);
+
+				} catch (MalformedURLException e) {
+					fail();
+				}
+			}
+		});
+		answeringSc.start();
+
+		// register the AskKB
+		HttpTester registerKb = new HttpTester(new URL(url + "/sc"), "POST",
+				"{\"knowledgeBaseId\": \"https://www.tno.nl/example/relationAsker\", \"knowledgeBaseName\": \"RelationAsker\", \"knowledgeBaseDescription\": \"A KB that asks for relations between people\", \"reasonerEnabled\" : true}",
+				Map.of("Content-Type", "application/json", "Accept", "*/*"));
+		registerKb.expectStatus(200);
+
+		// register the AskKI
+		HttpTester registerKiWithoutGapsEnabled = new HttpTester(new URL(url + "/sc/ki"), "POST",
+				"{\"knowledgeInteractionType\": \"AskKnowledgeInteraction\", \"knowledgeInteractionName\": \"askRelations\", \"graphPattern\": \"?a <http://example.org/isRelatedTo> ?b . ?a <http://example.org/isFatherOf> ?c .\", \"knowledgeGapsEnabled\": true}",
+				Map.of("Knowledge-Base-Id", "https://www.tno.nl/example/relationAsker", "Content-Type",
+						"application/json", "Accept", "*/*"));
+		registerKiWithoutGapsEnabled.expectStatus(200);
+
+		// fire the ask KI
+		HttpTester askKiWithoutGapsEnabled = new HttpTester(new URL(url + "/sc/ask"), "POST",
+				"{\"recipientSelector\": {\"knowledgeBases\": []}, \"bindingSet\": []}",
+				Map.of("Knowledge-Base-Id", "https://www.tno.nl/example/relationAsker", "Knowledge-Interaction-Id",
+						"https://www.tno.nl/example/relationAsker/interaction/askRelations", "Content-Type",
+						"application/json", "Accept", "*/*"));
+		var result = askKiWithoutGapsEnabled.getBody();
+		System.out.println("Result is:" + result);
+		assertTrue(result.contains("\"knowledgeGaps\":[[\"?a <http://example.org/isFatherOf> ?c\"]]"));
```

Comment:
For me this assertion failed 8 out of 10 times when running all REST API Unit Tests from Eclipse IDE. This was caused by a timing issue where the `relationProvider` KB was not ready yet, but the `relationAsker` already activated its `AskKnowledgeInteraction`. Fixed it using a `CountDownLatch` to synchronize between these two different threads.